cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java

```diff
@@ -125,19 +131,115 @@ private static void buildAngleRefs(Bus bus, Map<String, String> angleRefs, Cgmes
+        final String key;
+        final List<String> topologicalNodes;
+        // We need to export the load flow status for the island,
+        // from QoCDC 3.3.1 rule TIConvergenceStatMissing:
+        //      The cim:IdentifiedObject.description of cim:TopologicalIsland shall have one the
+        //      following string values: “converged” and “diverged” which represents the
+        //      convergence status of the cim:TopologicalIsland.
+        // Consider the island is converged unless we add a non-converged bus
+        String loadFlowStatus = CONVERGED;
+        final double maxPMismatchConverged;
+        final double maxQMismatchConverged;
+
+        private TopologicalIsland(String key, List<String> topologicalNodes, CgmesExportContext context) {
+            this.key = key;
+            this.topologicalNodes = topologicalNodes;
+            this.maxPMismatchConverged = context.getMaxPMismatchConverged();
+            this.maxQMismatchConverged = context.getMaxQMismatchConverged();
+        }
+
+        static TopologicalIsland fromSynchronousComponent(String key, CgmesExportContext context) {
+            return new TopologicalIsland(key, new ArrayList<>(), context);
+        }
+
+        static TopologicalIsland fromTopologicalNode(String topologicalNode, CgmesExportContext context) {
+            return new TopologicalIsland(topologicalNode, Collections.singletonList(topologicalNode), context);
+        }
+
+        void addNode(String topologicalNode, Bus bus, boolean updateLoadFlowStatus) {
+            topologicalNodes.add(topologicalNode);
+            if (updateLoadFlowStatus) {
+                updateLoadFlowStatus(bus);
+            }
+        }
+
+        void updateLoadFlowStatus(Bus bus) {
+            if (loadFlowStatus.equals(DIVERGED)) {
+                return;
+            }
+            if (!(isValidVoltage(bus.getV()) && isValidAngle(bus.getAngle()) && isInAccordanceWithKirchoffsFirstLaw(bus))) {
+                loadFlowStatus = DIVERGED;
+            }
+        }
+
+        boolean isValidVoltage(double v) {
+            return v >= 0.01;
+        }
+
+        boolean isValidAngle(double a) {
+            return Double.isFinite(a);
+        }
+
+        boolean isInAccordanceWithKirchoffsFirstLaw(Bus bus) {
+            // Instead of checking switch flows, we check that the corresponding bus view bus is balanced
+            // Maybe we will check the same bus view bus more than once
+            Optional<Bus> bus0 = getBusViewBus(bus);
+            if (bus0.isEmpty()) {
+                LOG.warn("Can not check if bus is in accordance with Kirchoff's first law. No BusView bus can be found for: {}", bus);
+                return false;
+            }
+            Complex sumTerminalFlows = bus0.get().getConnectedTerminalStream()
+                    .filter(this::isValidTerminal)
+                    .map(t -> new Complex(t.getP(), t.getQ()))
```

Comment:
@zamarrenolm just wondering what happens when t.getP() and/or t.getQ() is Double.NaN ?

Reply:
The sum will be NaN and the check against max p/q will fail, considering the bus as not according to Kirchoff's first law. I had to add conditions to ignore terminals for static var compensators with regulation OFF and busbar sections to avoid its NaN values.

Reply:
For the rest of situations where a terminal has `NaN`, I think it should be ok to consider the case as `diverged`.

Reply:
Ok so I disagree. A `NaN` terminal should be filtered and excluded from the sum. And I think you have to separate the P treatment that the Q treatment. Because we can have a NaN for P and not for Q. If something has be sum up because filtered, so we can say that it is a divergence. 
We have to be coherent, a NaN at terminal is exported as zero value.